### PR TITLE
Improves snowflake effect

### LINKED ISSUE #1562: Snow: Impove snowflake effect
http://php.quicoto.com/snow-html-canvas/, Demo: http://php.quicoto.com/examples/27/

This is more of the effect we'd like to see (simplified snowflakes, smaller).

Also, let's try 700 particles and update the background colour of the IA to `#D0D0D0`

https://duck.co/ia/view/snow

### LINKED ISSUE #1857: SimilarSites: Border should be re-added and set to transparent
We're about to make some internal changes to the spacing of the list elements. In order to preserve the intended spacing we need to change the CSS so that it sets `border-color: transparent` instead of removing the border.

------

IA Page: http://duck.co/ia/view/similar_sites

### PR DESCRIPTION
**Pull request by gokul1794**
This is how it looks like now, 
snowing
This should fix #1562

### CODE REVIEW

File: lib/DDG/Spice/Snow.pm [hunk 1]
```diff
@@ -17,7 +17,7 @@ attribution web => [ 'https://www.duckduckgo.com', 'DuckDuckGo' ],
             github => [ 'https://github.com/duckduckgo', 'DuckDuckGo'],
             twitter => ['http://twitter.com/duckduckgo', 'DuckDuckGo'];
 
-spice to => 'http://isitsnowingyet.org/api/check/$1/key/{{ENV{DDG_SPICE_SNOW_APIKEY}}}';
+spice to => 'https://duckduckgo.com/js/spice/snow/$1';
```
Comment (moollaza) [inline lib/DDG/Spice/Snow.pm]: This isn't necessary.

The `spice to` url is where the actual request is made to. The `duckduckgo.com/js/spice/snow/$1` call is re-written via NGXINX to the `spice to` url. This way the client makes a request to DDG and we in turn make a request to the API to maintain your privacy.

Please switch this back, otherwise the IA will break in production.

File: share/spice/similar_sites/similar_sites.css [hunk 2]
```diff
@@ -8,7 +8,7 @@
 
 .zci--similar_sites .c-list__items,
 .zci--similar_sites .c-list__item {
-    border: none;
+    border-color: transparent;
```
Comment (moollaza) [inline share/spice/similar_sites/similar_sites.css]: Whoops, looks like this belongs in another PR.

File: share/spice/snow/snow.js [hunk 3]
```diff
@@ -53,202 +53,104 @@ function nrio (api_result) {
     
     var $dom = Spice.getDOM('snow');
     $dom.css({
-        background: '#0C0F2B',
+        background: '#D0D0D0',
```
Comment (moollaza) [inline share/spice/snow/snow.js]: Let's stick with the original navy blue. It was much easier to read and see the snow against.

File: share/spice/snow/snow.js [hunk 4]
```diff
@@ -53,202 +53,104 @@ function nrio (api_result) {
...
+            };
+  })();
 
+  onResize();
+  window.addEventListener('resize', onResize, false);
 
-    makeItSnow();
+  masthead.appendChild(canvas);
+}
+ makeItSnow();
```
Comment (moollaza) [inline share/spice/snow/snow.js]: I think we need to check first if it's actually snowing?

Comment (moollaza) [inline share/spice/snow/snow.js]: Also, you don't want to fire the event here. You'll want to use the [onShow event](https://duck.co/duckduckhack/display_reference#Events).

File: share/spice/snow/snow.js [hunk 5]
```diff
@@ -53,202 +53,104 @@ function nrio (api_result) {
...
+  window.requestAnimFrame = (function(){
+    return  window.requestAnimationFrame       ||
+            window.webkitRequestAnimationFrame ||
+            window.mozRequestAnimationFrame    ||
+            function( callback ){
+              window.setTimeout(callback, 1000 / 60);
+            };
+  })();
 
+  onResize();
```
Comment (moollaza) [inline share/spice/snow/snow.js]: I believe this is why it's only firing on a re-size.

**Comment by gokul1794** [comment]
Made the necessary changes, looks like this now and triggers normally!
navybluebackground
Don't you think #2980b9 looks better?
lightbluesnow

File: share/spice/snow/snow.js [hunk 6]
```diff
@@ -45,11 +45,18 @@ function nrio (api_result) {
                 content: Spice.snow.content,
                 moreAt: true
             }
+        },
+        onShow: function(){
+         // add the snowflakes, if it's snowing
+         if(!api_result.is_snowing) { return; }
```
Comment (moollaza) [inline share/spice/snow/snow.js]: Nit: the closing brace should be on the line below:

```js
if (...) {
 ...
} else {
 ...
}
```

File: share/spice/snow/snow.css [hunk 7]
```diff
@@ -11,6 +11,6 @@
     font-weight: 900;
 }
 
-.zci--snow .zci__header {
-    color: inherit;
+.zci--snow.zci__header {
```
Comment (moollaza) [inline share/spice/snow/snow.css]: Looks like you can just remove this CSS rule

**Comment by moollaza** [comment]
@gokul1794 I'll let the design team decide on a color, but the Navy is definitely more legible.

**Comment by gokul1794** [comment]
@moollaza All right :)

**Comment by chrismorast** [comment]
Looking great @gokul1794 ! Let's stick with the Gray background color. However, can we change the subtext to # 666? The white text is a bit difficult to read. 

Any chance to get this up on a test machine to see the animation in action?

**Comment by gokul1794** [comment]
@chrismorast, Is this okay?

**Comment by chrismorast** [comment]
@gokul1794 , the default theme version is good. Any chance we can change the BG color for the dark theme to #292929?

**Comment by moollaza** [comment]
We need to use the 'tx-clr' and 'bg-clr' classes on the elements in order to make them look good across themes.

**Comment by moollaza** [comment]
They should be covered on the docs under Spice display, variants

**Comment by gokul1794** [comment]
@moollaza There's no mention of how to change the background color?

File: share/spice/snow/snow.css [hunk 8]
```diff
@@ -11,6 +11,6 @@
     font-weight: 900;
 }
 
-.zci--snow .zci__header {
-    color: inherit;
+.c-base__content {
```
Comment (moollaza) [inline share/spice/snow/snow.css]: whoops, you need to keep the `.zci--snow` here to namespace the CSS. Otherwise you're potentially going to change all the content on the page that matches your selector:

```css
.zci--snow .c-base__content {
   ...
}
```

File: share/spice/snow/snow.js [hunk 9]
```diff
@@ -45,11 +45,20 @@ function nrio (api_result) {
                 content: Spice.snow.content,
                 moreAt: true
             }
+        },
+        onShow: function(){
+         // add the snowflakes, if it's snowing
+         if(!api_result.is_snowing){
+             return; 
+             }
```
Comment (moollaza) [inline share/spice/snow/snow.js]: Indentation looks off here. Please make sure you're indented with soft-tabs (4 space)

File: share/spice/snow/snow.js [hunk 10]
```diff
@@ -57,198 +66,102 @@ function nrio (api_result) {
...
-        };
-
-        detectSize();
-
-        $(window).resize(detectSize);
-
-        if (!$dom.css('textShadow')) {
-            textShadowSupport = false;
-        }
+function makeItSnow() {
```
Comment (moollaza) [inline share/spice/snow/snow.js]: Please indent this code so it's inline with the above. 

Also, if you've borrowed any of this, I think we should leave a comment to give credit where it's due :)

**Comment by moollaza** [comment]
@gokul1794 to target the dark theme in the CSS please prefix your rule with `.dark-header`:

```css
.dark-header .zci--snow .foo {
   ...
}
```

**Comment by gokul1794** [comment]
@chrismorast,  Added, `#292929` This is how it looks.
@moollaza Made the changes
dark

**Comment by gokul1794** [comment]
@moollaza Our stylesheet?

**Comment by chrismorast** [comment]
Hey @gokul1794 , one more thing... please use "tx-clr--dk2" as the class for the subtext (No snow is falling, sigh). That way it will auto update to a lighter color for the dark theme. Thank you!

**Comment by gokul1794** [comment]
@chrismorast, Is this good enough?
subtext
subtext2

**Comment by gokul1794** [comment]
@moollaza Is this ready for merge?

**Comment by chrismorast** [comment]
@gokul1794 , yes, that's good. However, I notice now that the notch in the light gray version is not adjusted to match. I'm not sure we've accounted for that in updating the BG color. What would the lighter gray version look like with f2f2f2 as the BG (the default color)? Would it be too subtle? I apologize for the last minute change.

**Comment by gokul1794** [comment]
@chrismorast The problem with the default background color is that, the snowflakes aren't very visible!

**Comment by moollaza** [comment]
@gokul1794 I spoke with Chris and we agreed that we're going to revert back to the default background colour for the normal theme, because altering the colour creates issues with the separator in the More at link and the notch in the tab.

I'll merge this and update. :+1: 

------

BTW, in the future, please make sure you've formatted/indented your code properly. It looks like the indentation is off in a few places here. I'll fix :smiley:

**Comment by gokul1794** [comment]
@moollaza Sorry about that, I'll make sure it doesn't happen next time. Thanks! :+1:

**Comment by moollaza** [comment]
@gokul1794 its no problem :+1:  -- Let you editor do all that hard work -- I have a sublime text package that strips trailing whitespace and another that re-indents the code. Together they do all the heavy lifting :smile:

**Comment by moollaza** [comment]
Continued in #1993
